### ISSUE TITLE
`IsSubsetSet` (and others) change lists in-place

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#815]
Various functions in GAP (including `IsSubset`) can change functions in place, making them not a string rep, and mutable. This is probably a bad idea, and will certainly be a bad idea in HPC-GAP. I think the best fix is probably to change `PLAIN_LIST` to do a copy instead of an in-place modify (or at least, not in-place modify immutable variables).

```
gap> g := Immutable("E");
"E"
gap> IsMutable(g);
false
gap> IsStringRep(g);
true
gap> IsSubset("A", g);
false
gap> IsStringRep(g);
false
gap> IsMutable(g);
true
```

**Comment by alex-konovalov** [comment]
@ChrisJefferson The current behaviour is as follows:
```
gap> g := Immutable("E");
"E"
gap> IsMutable(g);
false
gap>  IsStringRep(g);
true
gap>  IsSubset("A", g);
false
gap> IsStringRep(g);
false
gap> IsMutable(g);
false
gap> 
```
so g is still immutable, but changes representation.

**Comment by fingolfin** [comment]
Yes, this minor improvement is due to b6cc57f8546c6ab64690f9b44fcbbaa6e242dee5

**Comment by fingolfin** [comment]
Here is another example, taken from #1095 and motivated by actual code that run into problems due to this bug:
```gap
gap> not IsMutable(CHARS_LALPHA); IsStringRep(CHARS_LALPHA);
true
true
gap> UniteSet("abc",CHARS_LALPHA);
gap> not IsMutable(CHARS_LALPHA); IsStringRep(CHARS_LALPHA);
true
false
```

**Comment by alex-konovalov** [comment]
This also causes problems for FR, see https://github.com/gap-packages/fr/issues/19

**Comment by fingolfin** [comment]
I fixed this not so long ago! Now, basically all places performing this inplace conversion via the kernel function `PLAIN_LIST` only do so for mutable lists, immutable lists effectively have "locked" representation...

... except for the kernel functions underlying `ConvertToVectorRep` and `ConvertToMatrixRep` which still call it on immutable objects -- and of course also the other way around, i.e. they convert stuff into compressed vectors/matrices even if it is immutable. This is a major design flaw of these functions, and one of the reasons I introduced `ImmutableVector` and improved `ImmutableMatrix`... I'd really like to purge / replaces all uses of these functions and related ones.

Anyway, I believe this issue can be closed now.
